packages/api/lib/errors.js

```diff
@@ -21,6 +27,14 @@ class AssociatedRulesError extends Error {
   }
 }
 
+class ApiCollisionError extends Error {
```

Comment:
Is this necessary vs using createErrorType?

Reply:
Necessary, maybe, preferable... probably?

createErrorType doesn't actually allow for a statement like `error instanceof ApiCollisionError` (if I'm recalling correctly, it's of type Error, I believe because it's anonymous).    We could argue that we should just be interrogating .name, but that gets rid of the inheritance behavior (e.g. subclasses of this error *as well as the parent* would also trigger this logic).

If that really grinds gears, I'm open to looking at an alternate approach.

Reply:
> createErrorType doesn't actually allow for a statement like error instanceof ApiCollisionError 

Are you sure about that? Because I've just implemented that in my other PR

https://github.com/nasa/cumulus/pull/1987/files#diff-bc96e8e9a9c08e28b506ef0387539a0fc75bb657b28dd974c4419818a5e72607R58

Reply:
Hmm, apparently not.     Agreed that's better given it works, although I'm concerned re: why I originally pulled it out of the generic generator setup.

d9e8cdbc0